src/content/cookbook/architecture/result.md

```diff
@@ -0,0 +1,460 @@
+  // ···
+
+  Future<Result<UserProfile>> getUserProfile() async {
+    // ···
+  }
+}
+```
+Instead of returning the `UserProfile` directly, 
+it returns a `Result` object containing a `UserProfile`.
+
+To facilitate using the `Result` class, 
+it contains two named constructors, `Result.ok` and `Result.error`. 
+Use them to construct the `Result` depending on desired output. 
+As well, capture any exceptions thrown by the code 
+and wrap them into the `Result` object.
+
+For example, here the `getUserProfile()` method 
+has been changed to use the `Result` class:
+
+<?code-excerpt "lib/main.dart (ApiClientService2)"?>
+```dart
+class ApiClientService {
+  // ···
+
+  Future<Result<UserProfile>> getUserProfile() async {
+    try {
+      final request = await client.get(_host, _port, '/user');
+      final response = await request.close();
+      if (response.statusCode == 200) {
+        final stringData = await response.transform(utf8.decoder).join();
+        return Result.ok(UserProfile.fromJson(jsonDecode(stringData)));
+      } else {
+        return Result.error(const HttpException('Invalid response'));
+      }
+    } on Exception catch (exception) {
+      return Result.error(exception);
+    } finally {
+      client.close();
+    }
+  }
+}
+```
+
+The original return statement was replaced 
+with a statement that returns the value using `Result.ok`. 
+The `throw HttpException()` 
+was replaced with a statement that returns `Result.error(HttpException())`,
+wrapping the error into a `Result`. 
+As well, the method is wrapped with a `try-catch` block
+to capture any exceptions thrown by the Http client 
+or the JSON parser into a `Result.error`.
+
+The repository class also needs to be modified, 
+and instead of returning a `UserProfile` directly, 
+now it returns a `Result<UserProfile>`.
+
+<?code-excerpt "lib/main.dart (getUserProfile1)" replace="/1//g"?>
+```dart
+Future<Result<UserProfile>> getUserProfile() async {
+  return await _apiClientService.getUserProfile();
+}
+```
+
+### Unwrapping the Result object
+
+Now the ViewModel doesn't receive the `UserProfile` directly, 
+but instead it receives a `Result` containing a `UserProfile`.
+
+This enforces the developer implementing the ViewModel 
+to unwrap the `Result` to obtain the `UserProfile`, 
+and avoids having uncaught exceptions.
+
+<?code-excerpt "lib/main.dart (UserProfileViewModel)"?>
+```dart
+class UserProfileViewModel extends ChangeNotifier {
+  // ···
+
+  UserProfile? userProfile;
+
+  Exception? error;
+
+  Future<void> load() async {
+    final result = await userProfileRepository.getUserProfile();
+    switch (result) {
+      case Ok<UserProfile>():
+        userProfile = result.value;
+      case Error<UserProfile>():
+        error = result.error;
+    }
+    notifyListeners();
+  }
+}
+```
+
+The `Result` class is implemented using a `sealed` class, 
+meaning it can only either be an `Ok` or an `Error`, 
+so this operation can be done using a switch case.
```

Comment:
```suggestion
The `Result` class is implemented using a `sealed` class, 
meaning it can only be of type `Ok` or `Error`. 
This allows the code to evaluate the result with a simple
switch statement.
```